libs/wire-subsystems/test/unit/Wire/MockInterpreters/UserSubsystem.hs

```diff
@@ -7,9 +7,22 @@ import Wire.API.User
 import Wire.UserKeyStore
 import Wire.UserSubsystem
 
-userSubsystemTestInterpreter :: [UserAccount] -> InterpreterFor UserSubsystem r
+userSubsystemTestInterpreter :: [ExtendedUserAccount] -> InterpreterFor UserSubsystem r
 userSubsystemTestInterpreter initialUsers =
   interpret \case
-    GetLocalUserAccountByUserKey localUserKey -> case (tUnqualified localUserKey) of
-      EmailKey _ email -> pure $ find (\u -> userEmail u.accountUser == Just email) initialUsers
+    GetExtendedAccountsBy (tSplit -> (_dom, getBy)) ->
+      pure $
+        filter
+          ( \u ->
+              mailKeyFrom u
+                `elem` getBy.getByEmail
+          )
+          initialUsers
     _ -> error $ "userSubsystemTestInterpreter: implement on demand"
+
+mailKeyFrom :: ExtendedUserAccount -> EmailKey
+mailKeyFrom acc =
+  case acc.account.accountUser.userIdentity of
+    Just (EmailIdentity mail) -> mkEmailKey mail
+    Just (SSOIdentity _ (Just mail)) -> mkEmailKey mail
+    _ -> error "Why are we testing users without emails for this?"
```

Comment:
is just the `Nothing` case missing here? (at least that's what I infer from the message?) if so, can we make that explicit?